### ISSUE TITLE
Test Airplay 2 with Denon and Sonos One

### ISSUE THREAD

**Issue by ejurgensen** [ejurgensen/forked-daapd#1161]
@jdtsmith hope it is ok if we bring the discussion about testing forked-daapd over here. The av_init_packet error sounds like ffmpeg is missing. If you are using Debian/Ubuntu, you can install the required dependencies [like this](https://github.com/ejurgensen/forked-daapd/blob/master/INSTALL.md#quick-version-for-debianubuntu-users).

If you already did that, and it still doesn't work, then please let me know which platform you are using.

**Comment by jdtsmith** [comment]
Sure.  I did install those all successfully on my Rasbian Buster system, and also install `ffmpeg`, but still no joy.  `configure` does report `checking for LIBAV... yes` just before:

```
checking for av_init_packet... no
configure: error: Unable to link function av_init_packet
```

**Comment by ejurgensen** [comment]
Do you have these packages installed: libavcodec-dev libavformat-dev libavfilter-dev libswscale-dev libavutil-dev

I do build on Raspbian Buster all the time, so it should work.

**Comment by jdtsmith** [comment]
OK that message was a red herring.   Looking in the config log I found complaints about `libmmal_core` etc. which are the actual cause of the compilation test for `av_init_packet` failing.  I searched and found `export LIBRARY_PATH=/opt/vc/lib`, which fixes it.  Possibly need to test for the `mmal` stuff in configure first?  Not sure why that's no on the linker's path.  Will take a quick test of sending now.

**Comment by ejurgensen** [comment]
Sounds pretty weird, hope you got it to work. Otherwise here is a deb for Raspbian Buster you can use: http://www.gyfgafguf.dk/temp/forked-daapd_27.2.110~rc3-3_armhf.deb

**Comment by ejurgensen** [comment]
Did you get around to testing with your Denon and Sonos?

**Comment by ejurgensen** [comment]
Closing, no feedback

**Comment by jdtsmith** [comment]
Apologies this took a bit of time to get it compiling locally.  I tried it out on master (commit 60b096c707141f27c065e03dda82ac7c500340ff), and it works great from the web interface, streaming local music to AP2 targets Sonos One and a Denon AVRX4400H (including both at the same time)!

For others trying to compile in a local test directory on Raspbian Buster to try this out before fully deploying, I succeeded doing so with:

```
export LIBRARY_PATH=/opt/vc/lib
./configure --enable-preferairplay2 --prefix=/home/pi/fdtmp/usr --sysconfdir=/home/pi/fdtmp/etc --localstatedir=/home/pi/fdtmp/var
make
make install
cd /home/pi/
fdtmp/usr/sbin/forked-daapd --foreground --config=fdtmp/etc/forked-daapd.conf -w fdtmp/usr/share/forked-daapd/htdocs/
```

also editing the `forked-daapd.conf` file to point everything to `/home/pi/fdtmp`.
